feat(akka/bbb-web): persist plugin errors from back-end to be logged into the front-end

GuiLeme opened the pull request:
### What does this PR do?

This PR basically persists plugin loading errors from back-end (bbb-web/akka) so that they will be logged on the front-end. 

### Closes Issue(s)

Closes #22746

### Motivation

Since the plugin loading errors are distributed across some of the services we have, it was a bit difficult to discover where exactly a plugin failed to load (that could be in `akka` or `bbb-web`).

With this PR, not only we'll be able to know what was wrong (in order to fix it) via `errorMessage`, but also, where the error happened, so that we can seek more details already knowing where to search, via `loadingErrorSource` (this can be `bbb-web` or `bbb-akka-apps`); 

### How to test

I propose testing all the handled errors and try to break the plugin loading process in a creative manner (if you find one, please comment below so we can fix it promptly) - All of which should be logged into the client console.

GuiLeme committed:
[plugin-loading-error] - persist all loading error from back-end (akka and bbb-web) to be logged in the front-end

GuiLeme committed:
[plugin-loading-error] update version in properties

gustavotrott commented:
I'm not sure we should log the manifest URL for all clients, as it might not be public. It may be better to only expose the `javascriptEntry` for clients.
In this case, we could log a more generic message like "I/O error when fetching a plugin manifest URL" on the client side, while keeping the full URL logged on the backend.

<img width="3151" height="435" alt="image" src="https://github.com/user-attachments/assets/f27ff412-6fce-4f67-9964-0470c6a2c900" />

gustavotrott commented on bigbluebutton-html5/imports/ui/components/plugins-engine/manager.tsx:
```diff
@@ -53,6 +63,22 @@ const PluginsEngineManager = (props: PluginsEngineManagerProps) => {
+  useEffect(() => {
+    if (failedPlugins && failedPlugins.length > 0) {
+      failedPlugins.forEach((p) => {
+        logger.error({
```
I suggest logging it as `debug` instead of `error`.
Using `error` would cause every client to send the log to the server, which could lead to flooding with the same repeated message.
The error is already being logged on the `bbb-web` or `akka-apps` side.

gustavotrott commented:
When I set `name: ""` in the manifest, the log says the plugin was loaded but it actually doesn't load.

<img width="1900" height="297" alt="image" src="https://github.com/user-attachments/assets/b499bf8c-51af-40ac-9edc-0fd09aa55a7c" />

gustavotrott reviewed (changes requested):
Just small suggestions.
Other than that it's working properly!

GuiLeme committed:
[plugin-loading-error] resolve conflicts when merging upstream

GuiLeme committed:
[plugin-loading-error] resolve missing pluginName; error -> debug level; hide plugin manifest URL

GuiLeme commented:
@gustavotrott 
> It may be better to only expose the javascriptEntry for clients.

I'm afraid that is not possible. See, the point of logging the plugin manifest URL was to have an identifier because we can't either fetch the manifest URL or parse its result, so since the  `javascriptEntrypointUrl` is located within the manifest content, we cannot get it as well.

But I feel you're concern is valid, so, I removed the manifest URL from the front-end, and added a message to encourage the user to see bbb-web logs for more information when the name is not there.

gustavotrott commented on akka-bbb-apps/src/main/scala/org/bigbluebutton/core/models/Plugins.scala:
```diff
@@ -216,44 +236,87 @@ object PluginModel {
-      PluginDAO.insert(meetingId, plugin.manifest.content.name, plugin.manifest.content.javascriptEntrypointUrl,
-        plugin.manifest.content.javascriptEntrypointIntegrity.getOrElse(""), plugin.manifest.content.localesBaseUrl)
+    instance.plugins.foreach { case (pluginNameRaw, plugin) =>
+      val pluginName = if (plugin.manifest.url == pluginNameRaw) "NOT-USED" else pluginNameRaw
```
```suggestion
      val pluginName = if (plugin.manifest.url == pluginNameRaw) "unidentified-plugin" else pluginNameRaw
```

The name `NOT-USED` is a bit cofusing:
<img width="1859" height="478" alt="image" src="https://github.com/user-attachments/assets/355d3ef4-4ad7-496d-8661-50a3a021d188" />

GuiLeme committed:
[plugin-loading-error] rename unidentified plugin when no pluginName is found

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

gustavotrott reviewed (approved):
lgtm, well done!
